File: SUPPORTING FILES/SPCHOL200/Completed/CS/Ex1/SPCHOL200-Ex1/VisualWebPart1/VisualWebPart1UserControl.ascx.cs
//--------------------------------------------------------------------------------
// This file is a "Sample" from the SharePoint Foundation 2010
// Samples
//
// Copyright (c) Microsoft Corporation. All rights reserved.
//
// This source code is intended only as a supplement to Microsoft
// Development Tools and/or on-line documentation.  See these other
// materials for detailed information regarding Microsoft code samples.
// 
// THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//--------------------------------------------------------------------------------

using System;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using Microsoft.SharePoint.Linq;
using Microsoft.SharePoint;
using System.Linq;


namespace SPCHOL200_Ex1.VisualWebPart1
{
    public partial class VisualWebPart1UserControl : UserControl
    {
        protected void Page_Load(object sender, EventArgs e)
        {
            var dc = new SPLinqDataContext(SPContext.Current.Web.Url);

            var Employees = dc.GetList<EmployeesItem>("Employees");

            var empQuery = from emp in Employees
                           where emp.Project.DueDate <
                           DateTime.Now.AddMonths(6)
                           select new
                           {
                               emp.Title,
                               emp.JobTitle,
                               ProjectTitle = emp.Project.Title,
                               DueDate =
                               emp.Project.DueDate.Value.ToShortDateString()
                           };

            spGridView.DataSource = empQuery;
            spGridView.DataBind();


        }
    }
}

File: SUPPORTING FILES/SPCHOL200/Completed/VB/Ex1/SPCHOL200-Ex1/VisualWebPart1/VisualWebPart1UserControl.ascx.vb
'--------------------------------------------------------------------------------
' This file is a "Sample" from the SharePoint Foundation 2010
' Samples
'
' Copyright (c) Microsoft Corporation. All rights reserved.
'
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'--------------------------------------------------------------------------------

Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts

Imports Microsoft.SharePoint.Linq
Imports Microsoft.SharePoint
Imports System.Linq

Partial Public Class VisualWebPart1UserControl
    Inherits UserControl

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load

        Dim dc As New SPLinqDataContext(SPContext.Current.Web.Url)

        Dim Employees As EntityList(Of EmployeesItem) = _
        dc.GetList(Of EmployeesItem)("Employees")

        Dim empQuery = From emp In Employees _
            Where emp.Project.DueDate < DateTime.Now.AddMonths(6) _
            Select emp.Title, emp.JobTitle, ProjectTitle = emp.Project.Title, _
            DueDate = emp.Project.DueDate.ToString()

        spGridView.DataSource = empQuery
        spGridView.DataBind()

    End Sub

End Class
